backend/routes/monitoring.py

```diff
@@ -192,3 +199,86 @@ def update_synthese_api(module_code):
         .get()
         .process_synthese(process_module=True)
     )
+# export add mje
+# export all observations
+@blueprint.route('/module/<module_code>/<type>/<method>/<jd>', methods=['GET'])
+@check_cruved_scope_monitoring('R', 1)
+def export_all_observations(module_code, type, method,jd):
+    """
+    Export all the observations made on a site group.
+    Following formats are available:
+    * csv
+    * geojson
+    * shapefile
+    """
+    
+    view = GenericTableGeo(
+        tableName="v_export_" + module_code.lower()+"_"+method, 
+        schemaName="gn_monitoring", 
+        engine=DB.engine
+
+    )
+    columns = view.tableDef.columns
+    q = DB.session.query(*columns)
+    #data = q.all()
+    #----------------------------
+    data = DB.session.query(*columns).filter(columns.id_dataset == jd).all()
+    #-------------------------------------
+
+    filename = module_code+"_"+method+"_"+dt.datetime.now().strftime("%Y_%m_%d_%Hh%Mm%S")
+
+    if type == 'csv':
+        return to_csv_resp(
+            filename,
+            data=serializeQuery(data, q.column_descriptions),
+            separator=";",
+            columns=[db_col.key for db_col in columns if db_col.key != 'geom'], # Exclude the geom column from CSV
+        )
+    elif type == 'geojson':
+        results = FeatureCollection([view.as_geofeature(d, columns=columns) for d in data])
+        return to_json_resp(results, as_file=True, filename=filename, indent=4, extension='geojson')
+    elif type == 'shp':
+        try:
+            fm.delete_recursively(
+                str(ROOT_DIR / "backend/static/shapefiles"), excluded_files=[".gitkeep"]
+            )
+            db_cols = [db_col for db_col in view.db_cols if db_col.key in columns]
+            dir_path = str(ROOT_DIR / "backend/static/shapefiles")
+            view.as_shape(
+                db_cols=db_cols, data=data, dir_path=dir_path, file_name=filename
+            )
+            return send_from_directory(dir_path, filename + ".zip", as_attachment=True)
+
+        except GeonatureApiError as e:
+            return render_template(
+                "error.html",
+                error=str(e),
+                redirect=current_app.config["URL_APPLICATION"] + "/#/cmr",
```

Comment:
la redirection va vers le module `cmr`

Reply:
c’une fonction pour faire plusieurs type d’export qu’on a developpé conjoitement avec le module CMR. 
Por le monitoring on a besoin uniquement de type CSV cad on garde uniquement le morceau de code suivant :
    if type == 'csv':
        return to_csv_resp(
            filename,
            data=serializeQuery(data, q.column_descriptions),
            separator=";",
            columns=[db_col.key for db_col in columns if db_col.key != 'geom'], # Exclude the geom column from CSV
        )
    else:
        raise NotFound("type export not found")

Reply:
c'est fait